src/components/network-map-tab.js

```diff
@@ -194,78 +183,123 @@ export const NetworkMapTab = ({
+        (foundEquipmentPositions, allEquipments, fetchEquipmentCB) => {
+            let notFoundEquipmentsIds = [];
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+            allEquipments.forEach((s) => {
+                if (!foundEquipmentsIds.includes(s.id)) {
+                    notFoundEquipmentsIds.push(s.id);
+                }
+            });
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+            setWaitingLoadGeoData(true);
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
     const reloadMapGeoData = useCallback(() => {
-        console.info(`Loading geo data of study '${studyUuid}'...`);
-        const substationPositions = fetchSubstationPositions(
-            studyUuid,
-            currentNode?.id
-        );
-        const linePositions = lineFullPath
-            ? fetchLinePositions(studyUuid, currentNode?.id)
-            : [];
-        setWaitingLoadGeoData(true);
-
-        Promise.all([substationPositions, linePositions])
-            .then((values) => {
-                const newGeoData = new GeoData();
-                newGeoData.setSubstationPositions(values[0]);
-                newGeoData.setLinePositions(values[1]);
-                setGeoData(newGeoData);
-                setWaitingLoadGeoData(false);
-            })
-            .catch(function (error) {
-                console.error(error.message);
-                setWaitingLoadGeoData(false);
-                setErrorMessage(
-                    intlRef.current.formatMessage(
-                        { id: 'geoDataLoadingFail' },
-                        { studyUuid: studyUuid }
-                    )
+        if (studyUuid && currentNode) {
+            if (
+                geoData &&
+                (geoData.substationPositionsById.size > 0 ||
+                    geoData.linePositionsById.size > 0)
+            ) {
+                const missingSubstationPositions =
+                    getMissingEquipmentsPositions(
+                        geoData.substationPositionsById,
+                        network.getSubstations(),
+                        fetchSubstationPositionsByIds
+                    );
+
+                const missingLinesPositions = getMissingEquipmentsPositions(
+                    geoData.linePositionsById,
+                    network.getLines(),
+                    fetchLinePositionsByIds
                 );
-            });
-        dispatch(resetMapReloaded());
+
+                Promise.all([
+                    missingSubstationPositions,
+                    missingLinesPositions,
+                ]).then((positions) => {
+                    geoData.addSubstationPositions(positions[0]);
```

Comment:
Maybe deal with the case when `positions[0]` and `positions[1]` are both empty ?

Reply:
It's a light benefice, but ok, done